packages/hint-doctype/src/hint.ts

```diff
@@ -0,0 +1,125 @@
+            return new RegExp(/(<!doctype\s(html)\s*?)(>)(.+)?/, flags ? flags : 'g');
+        };
+
+        const checkDoctypeIsValid = async (resource: string, element: IAsyncHTMLElement | null, content: string): Promise<boolean> => {
+            debug(`Checking if the doctype is valid.`);
+
+            const matched = content.match(doctypeRegexFactory('gi'));
+
+            if (!matched || matched.length < 1) {
+                await context.report(resource, element, `The file does not contain a valid doctype tag`);
+
+                return false;
+            }
+
+            return true;
+        };
+
+        const checkDoctypeFirstLine = async (resource: string, element: IAsyncHTMLElement | null, content: string): Promise<void> => {
+            debug(`Checking if the doctype is in the first line.`);
+
+            const firstLine = content.split(/\r|\n/)[0];
+            const matched = firstLine.match(doctypeRegexFactory('gi'));
+
+
+            if (!matched || matched.length < 1) {
+                await context.report(resource, element, `The first line does not contain a valid doctype tag.`);
+
+                return;
+            }
+
+            if (matched) {
+                const cleaned = matched[0].trim().split('');
+
+                if (cleaned[cleaned.length-1] !== '>') {
+                    await context.report(resource, element, `There is additional information on the line with the doctype tag`);
+
+                    return;
+                }
+            }
+        };
+
+        const checkDoctypeLowercase = async (resource: string, element: IAsyncHTMLElement | null, content: string): Promise<void> => {
+            debug(`Checking that the doctype is in lowercase`);
+
+            const matched = content.match(doctypeRegexFactory());
+
+            if (!matched) {
+                await context.report(resource, element, `The doctype should be in lowercase`);
+
+                return;
+            }
+        };
+
+        const checkNoDuplicateDoctype = async (resource: string, element: IAsyncHTMLElement | null, content: string): Promise<void> => {
+            debug(`Checking that there is only one doctype tag in the document`);
+
+            const matched = content.match(doctypeRegexFactory('gi'));
+
+            if (matched && matched.length > 1) {
+                await context.report(resource, element, `There is more than one doctype tag in the document`);
+
+                return;
+            }
+        };
+
+        const onFetchEndHTML = async (fetchEnd: FetchEnd): Promise<void> => {
+            const { resource, element, response } = fetchEnd;
+
+            if (!response || !response.body || !response.body.content) {
+                await context.report(resource, element, 'Content has no body');
+
+                return;
+            }
+
+            const { body } = response;
+            const { content } = body;
+
+            /*
+             * If doctype is not valid, don not run more tests
```

Comment:
Nit: You can collapse this into one line comment: `// If DOCTYPE is not valid, do not run more tests` . Also you have a typo in
> do~~n~~ not